File: intern-backend/src/main/java/com/internBackend/timeoff/service/TimeoffService.java
package com.internBackend.timeoff.service;

import com.internBackend.timeoff.entity.Timeoff;

import java.util.List;

public interface TimeoffService {
    List<Timeoff> fetchAllTimeoffs();
    List<Timeoff> getAllTimeoffsByEmployeeId(Long employeeId);
    Long countAllTimeoffs();
    Long countAllApprovedTimeoffs();
    Long countAllApprovedTimeoffsByEmployeeId(Long employeeId);
    Long countUsedTimeoffsByEmployeeId(Long employeeId);


    Integer getRemainingTimeoffByEmployeeId(Long employeeId);

    Timeoff fetchTimeoffById(Long id);
    Timeoff createTimeoff(Timeoff timeoff);
    Timeoff updateTimeoff(Timeoff timeoff);
    String deleteTimeoff(Timeoff timeoff);
    Timeoff changeTimeoffStatus(Long id);
    }

File: intern-frontend/src/app/timeoff/service/timeoff.service.ts
// src/app/services/timeoff.service.ts

import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {Observable} from 'rxjs';
import {Timeoff} from '../model/timeoff.model';

@Injectable({
  providedIn: 'root',
})
export class TimeoffService {
  private baseUrl: string = 'http://localhost:8085/api/v1/timeoffs';

  constructor(private http: HttpClient) {
  }


  fetchAllTimeoffs(): Observable<Timeoff[]> {
    return this.http.get<Timeoff[]>(`${this.baseUrl}`);
  }

  getAllTimeoffsByEmployeeId(employeeId: number): Observable<Timeoff[]> {
    return this.http.get<Timeoff[]>(`${this.baseUrl}/employee/${employeeId}`);
  }

  createTimeoff(data: Timeoff): Observable<Timeoff> {
    return this.http.post<Timeoff>(`${this.baseUrl}`, data);
  }


  updateTimeoff(data: Timeoff): Observable<Timeoff> {
    return this.http.put<Timeoff>(`${this.baseUrl}/${data.id}`, data);
  }


  deleteTimeoff(id: number): Observable<{ message: string }> {
    return this.http.delete<{ message: string }>(`${this.baseUrl}/${id}`);
  }

  changeTimeoffStatus(id: number): Observable<Timeoff> {
    return this.http.get<Timeoff>(`${this.baseUrl}/changeTimeoffStatus/${id}`);
  }

}
